src/components/SearchPost.tsx

```diff
@@ -0,0 +1,62 @@
+import { Checkbox, Input, Stack, Text } from "@chakra-ui/react";
+import React, { useState } from "react";
+import { BlogPostCategory } from "../types/blogPost";
+
+interface SearchPostProps {
+  handleSearchCategory: (category: BlogPostCategory[]) => void;
+  handleSearchWord: (word: string) => void;
+  handlePaginationPageChange: (paginationPage: number) => void;
```

Comment:
Filters component should know nothing about pagination. You can have page reset logic in the parent component.

```
const Homepage = () => {
  const [postListFilters, setPostListFilters] = useState<PostListFiltersValues>({...})
  const [page, setPage]  = useState(0);

  const handlePostListFiltersChange = (filters: PostListFiltersValues) => {
   setPostListFilters(filters);
   setPage(0);
  }

 return <>
...
<PostListFilters filters={postListFilters} onFiltersChange={handlePostListFiltersChange} />
...
<>
}

```